internal/clients/http/azure/create_vms.go

```diff
@@ -0,0 +1,50 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/google/uuid"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+)
+
+func (c *client) CreateVMs(ctx context.Context, vmParams clients.AzureInstanceParams, amount int64, vmNamePrefix string) ([]*string, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "CreateVMs")
+	defer span.End()
+
+	logger := logger(ctx)
+	logger.Debug().Msgf("Started creating %d Azure VM instances", amount)
+
+	vmIds := make([]*string, amount)
+	var resumeTokens []string
```

Comment:
Why vmIds is preallocated while tokens are not? They should be both the same, I mean if you skip a VM because of error, then it makes sense to not to preallocate vmids.

Also, vmIds are pointers so they may contain nils. Why is that? If I launch 5 instances and one is skipped, receiving slice of 4 values and one randomly placed nil is not a great experience.

I suggest to make this consistent and when VMs were skipped for some reason, return a slice that has a lower length. We cannot do much about it, there is no transactions or anything :-)

Reply:
Pushed a new commit for easier review with strongly typing vmIDs and preallocated resumeTokens.